components/plurals/src/lib.rs

```diff
@@ -875,3 +875,97 @@ where
             .unwrap_or(end)
     }
 }
+
+#[derive(Debug)]
+#[non_exhaustive]
+/// A bag of values for different plural cases.
+pub struct PluralElements<'a, T: ?Sized> {
+    /// The value used for the [`PluralCategory::Zero`] case
+    pub zero: Option<&'a T>,
+    /// The value used for the [`PluralCategory::One`] case
+    pub one: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Two`] case
+    pub two: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Few`] case
+    pub few: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Many`] case
+    pub many: Option<&'a T>,
+    /// The value used for the [`PluralCategory::Other`] case
+    pub other: &'a T,
+    /// The value used when the [`PluralOperands`] are exactly 0.
+    pub explicit_zero: Option<&'a T>,
+    /// The value used when the [`PluralOperands`] are exactly 1.
+    pub explicit_one: Option<&'a T>,
+}
+
+impl<'a, T: ?Sized + PartialEq> PluralElements<'a, T> {
+    /// Creates a new [`PluralElements`] with the given default value.
+    pub fn new(other: &'a T) -> Self {
+        Self {
+            other,
+            zero: None,
+            one: None,
+            two: None,
+            few: None,
+            many: None,
+            explicit_zero: None,
+            explicit_one: None,
+        }
+    }
+
+    /// Sets the value for [`PluralCategory::Zero`].
+    pub fn with_zero_value(self, zero: Option<&'a T>) -> Self {
```

Comment:
Thought: these fns are fine but the struct fields are pub so we don't really need the setters, and they could be distracting

Reply:
I think @robertbastian  wants to check for each element if it is equal to `other` or not. 

The only think with this approach, that it is not general enough, I am not sure if CLDR decided to fall back to another category, for example `one`

Reply:
the setters allow constructing a `PluralElements` as a single expression (`PluralElements::new("bar").with_zero_value(Some("foo"))`). Otherwise you'd always need a block (`{ let mut temp = PluralElements::new("bar"); temp.zero_value = Some("foo"); temp }`)